File: app/src/main/java/ru/tajwid/app/ui/adapter/TimeZoneAdapter.java
package ru.tajwid.app.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import ru.tajwid.app.R;
import ru.tajwid.app.ui.activity.TimeZoneLearning;

public class TimeZoneAdapter extends RecyclerView.Adapter<TimeZoneAdapter.MyViewHolder> {

    Context context;

    ArrayList<TimeZoneLearning> list;

    public TimeZoneAdapter(Context context, ArrayList<TimeZoneLearning> list) {
        this.context = context;
        this.list = list;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.item_timelearning, parent, false);
        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        TimeZoneLearning timeZoneLearning = list.get(position);
        holder.name.setText(timeZoneLearning.getName());
        holder.data.setText(timeZoneLearning.getData());

//        holder.url.setText(timeZoneLearning.getUrl());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextView name, data, url;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            name = itemView.findViewById(R.id.username);
            data = itemView.findViewById(R.id.userdata);

//            url=itemView.findViewById(R.id.url);
        }
    }


}

File: app/src/main/java/ru/tajwid/app/ui/adapter/TimeZoneAdapter.kt
package ru.tajwid.app.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.view.isVisible
import androidx.recyclerview.widget.RecyclerView
import ru.tajwid.app.R
import ru.tajwid.app.content.data.Schedule
import ru.tajwid.app.ui.adapter.TimeZoneAdapter.MyViewHolder

class TimeZoneAdapter : RecyclerView.Adapter<MyViewHolder>() {

    private var list: List<Schedule> = emptyList()
    private var price: String = ""

    fun setData(list: List<Schedule>, price: String) {
        this.list = list
        this.price = price
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val v =
            LayoutInflater.from(parent.context).inflate(R.layout.item_timelearning, parent, false)
        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val timeZoneLearning = list[position]
        holder.bind(timeZoneLearning)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val teacher: TextView by lazy { itemView.findViewById(R.id.teacher) }
        val time: TextView by lazy { itemView.findViewById(R.id.time) }
        val weekDay: TextView by lazy { itemView.findViewById(R.id.week_day) }
        val priceText: TextView by lazy { itemView.findViewById(R.id.price) }

        fun bind(schedule: Schedule) {
            teacher.text =
                itemView.context.getString(R.string.title_item_teacher_name, schedule.teacher)
            time.text = schedule.time
            weekDay.text = schedule.day
            priceText.text = itemView.context.getString(R.string.price_placeholder, price)
            priceText.isVisible = price.isNotBlank()
        }
    }
}